### PR TITLE
[All hosts] (AppSource) Update Edge Legacy validation guidance

### PR DESCRIPTION
**Pull request by samantharamon**
AppSource no longer rejects add-in submissions that don't support Microsoft Edge Legacy (WebView1, EdgeHTML-based); it now only provides a warning to the submitter.

### CODE REVIEW

File: docs/concepts/browsers-used-by-office-web-add-ins.md
Comment (AlexJerabek) [inline docs/concepts/browsers-used-by-office-web-add-ins.md:61]: You already caveat (EdgeHTML) earlier in the article. Do you need to do it again here, or would "Microsoft Edge Legacy" be sufficient? If you do change it here, please change it later in the article as well.

File: docs/concepts/browsers-used-by-office-web-add-ins.md
Comment (AlexJerabek) [inline docs/concepts/browsers-used-by-office-web-add-ins.md:107]: Possessives on brand names sometimes don't translate well, so it's best to avoid them when possible.
```suggestion
Office Add-ins don't support Service Workers when the Microsoft Edge Legacy WebView, [EdgeHTML](https://en.wikipedia.org/wiki/EdgeHTML), is used. They're supported with the [Chromium-based Edge WebView2](/microsoft-edge/hosting/webview2).
```

File: docs/develop/support-ie-11.md
Comment (AlexJerabek) [inline docs/develop/support-ie-11.md:2]: I think this is fine as is, but did you consider titling this page "Support older Microsoft browsers and Office versions"?

File: docs/develop/support-ie-11.md
Comment (AlexJerabek) [inline docs/develop/support-ie-11.md:11]: I know this is acting like a header, but I think the verb helps readability.
```suggestion
> **Internet Explorer and Microsoft Edge Legacy are still used in Office Add-ins**
```

File: docs/develop/support-ie-11.md
Comment (AlexJerabek) [inline docs/develop/support-ie-11.md:13]: Brevity suggestion.
```suggestion
> Some combinations of platforms and Office versions, including perpetual versions through Office 2019, still use the webview controls that come with Internet Explorer 11 and Microsoft Edge Legacy (EdgeHTML-based) to host add-ins, as explained in [Browsers used by Office Add-ins](../concepts/browsers-used-by-office-web-add-ins.md). We recommend (but don't require) that you support these combinations, at least in a minimal way, by providing users of your add-in a graceful failure message when your add-in is launched in these webviews. Keep these additional points in mind:
```

File: docs/develop/support-ie-11.md
Comment (AlexJerabek) [inline docs/develop/support-ie-11.md:15]: Can someone have Edge Legacy as a browser anymore? These articles are written from a browser-centric perspective, even though it's really the Office version that's dictating the webview.

File: docs/develop/support-ie-11.md
Comment (AlexJerabek) [inline docs/develop/support-ie-11.md:16]: Since this is two sentences, I'd favor the period over the semicolon. Feel free to disagree.
```suggestion
> - AppSource still tests for combinations of platform and Office *desktop* versions that use Internet Explorer and Microsoft Edge Legacy. However, it only issues a warning when the add-in doesn't support these browsers. The add-in isn't rejected by AppSource.
```

File: docs/develop/support-ie-11.md
Comment (AlexJerabek) [inline docs/develop/support-ie-11.md:25]: Why tabs instead of H2 sections?

File: docs/develop/support-ie-11.md
Comment (AlexJerabek) [inline docs/develop/support-ie-11.md:21]: I think you've defined the"Edge-based" bit earlier in the article.
```suggestion
If you plan to support older versions of Windows and Office, your add-in must work in the embeddable browser control that is based on Internet Explorer 11 (IE11) or Microsoft Edge Legacy. For information about which combinations of Windows and Office use these browser controls, see [Browsers used by Office Add-ins](../concepts/browsers-used-by-office-web-add-ins.md).
```

File: docs/develop/support-ie-11.md
Comment (AlexJerabek) [inline docs/develop/support-ie-11.md:77]: Should this section go earlier in the article?

File: docs/develop/support-ie-11.md
Comment (AlexJerabek) [inline docs/develop/support-ie-11.md:82]: Shouldn't this be `===`?

File: docs/develop/support-ie-11.md
Comment (AlexJerabek) [inline docs/develop/support-ie-11.md:92]: Same equal question?

File: docs/testing/test-debug-office-add-ins.md
Comment (AlexJerabek) [inline docs/testing/test-debug-office-add-ins.md:19]: This IMPORTANT note is getting huge. Should it become an H2 section?

File: docs/testing/test-debug-office-add-ins.md
Comment (AlexJerabek) [inline docs/testing/test-debug-office-add-ins.md:26]: Same question here as earlier about the availability of old Edge as a browser.

Comment (AlexJerabek) [inline docs/testing/test-debug-office-add-ins.md:26]: I think we should link to the issues list, rather than the new issue template. That way, the user can check for duplicates.
```suggestion
> AppSource doesn't use Internet Explorer or Microsoft Edge Legacy (EdgeHTML) to test add-ins in Office on the web. However, if a significant number of your users will use Edge Legacy to open Office on the web, then you should test with it. (Office on the web will not open in Internet Explorer, so you can't, and don't need to, test Office on the web with Internet Explorer.) For more information, see [Support Internet Explorer 11 and Microsoft Edge Legacy](../develop/support-ie-11.md) and [Troubleshooting Microsoft Edge issues](../concepts/browsers-used-by-office-web-add-ins.md#troubleshoot-microsoft-edge-issues). Office still supports these browsers for add-in runtimes, so if you think you've encountered a bug in how add-ins run in them, please create an issue for the [office-js](https://github.com/OfficeDev/office-js/issues) repo.
```

File: docs/testing/test-debug-office-add-ins.md
Comment (AlexJerabek) [inline docs/testing/test-debug-office-add-ins.md:30]: Consider breaking up this first sentence to make it easier to parse.
```suggestion
> Some Office versions on Windows still use the webview controls that come with Internet Explorer and Microsoft Edge Legacy. AppSource tests whether your add-in supports these browsers. If your add-in doesn't support these browsers, AppSource only issues a warning and doesn't reject your add-in. In this instance, we recommend configuring a graceful failure message on your add-in for a smoother user experience. For further guidance, see [Support Internet Explorer 11 and Microsoft Edge Legacy](../develop/support-ie-11.md).
```

**Review by AlexJerabek — COMMENTED** [review]
Thanks for putting this together @samantharamon. I have a few questions before signing off, but looks good overall.

File: docs/concepts/browsers-used-by-office-web-add-ins.md
Comment (samantharamon) [reply docs/concepts/browsers-used-by-office-web-add-ins.md:61]: The EdgeHTML reference first appears in the Important block. My thought is that if the user is directly linked to the "Perpetual versions" section or somehow skips the Important block, they would have another way to differentiate between the old and new Edge versions through that EdgeHTML reference.

File: docs/develop/support-ie-11.md
Comment (samantharamon) [reply docs/develop/support-ie-11.md:2]: That's a great suggestion and would definitely help with updating this article in the future. I'll make the change.

File: docs/develop/support-ie-11.md
Comment (samantharamon) [reply docs/develop/support-ie-11.md:15]: Good point. Even if Edge Legacy were somehow still installed on older Windows OS versions, such as Windows 7 and 8/8.1, users would be prompted to update to the Chromium-based version to be able to use the browser. I'll update line 15 (and the other article where this is referenced) to include Edge Legacy.

File: docs/develop/support-ie-11.md
Comment (samantharamon) [reply docs/develop/support-ie-11.md:21]: Similar to my earlier comment about this being the first EdgeHTML reference outside the Important note block. Although, I'm open to changing this if we think the note block mention is sufficient.

File: docs/develop/support-ie-11.md
Comment (samantharamon) [reply docs/develop/support-ie-11.md:25]: I figured it would be easier to reach the Edge Legacy sections (if that's what the user is after) without having to scroll through the IE sections.

File: docs/develop/support-ie-11.md
Comment (samantharamon) [reply docs/develop/support-ie-11.md:77]: That makes sense. I'll move it up.

File: docs/develop/support-ie-11.md
Comment (samantharamon) [reply docs/develop/support-ie-11.md:82]: No. A -1 value indicates that the specified browser isn't in use. Because we can't guarantee the index of the browser in the returned string, it's safer to do this check instead of using a specific value.

File: docs/develop/support-ie-11.md
Comment (samantharamon) [reply docs/develop/support-ie-11.md:92]: See earlier comment.

File: docs/testing/test-debug-office-add-ins.md
Comment (samantharamon) [reply docs/testing/test-debug-office-add-ins.md:19]: I was going back and forth on this. Now that I look at it again, the note does take up majority of the article space now. I'll move this into its own section.

File: docs/develop/support-ie-11.md
Comment (AlexJerabek) [reply docs/develop/support-ie-11.md:82]: Ah, gotcha. I had to re-read that to make sense of the double negative.

File: docs/testing/test-debug-office-add-ins.md
Comment (AlexJerabek) [inline docs/testing/test-debug-office-add-ins.md:21]: This is probably out of scope for this PR, but this segment here now feels redundant. The prior section talks about cross-plat. Maybe it's just me.

File: docs/testing/test-debug-office-add-ins.md
Comment (AlexJerabek) [inline docs/testing/test-debug-office-add-ins.md:27]: Consider putting the "learn more" link at the end of the paragraph.
```suggestion
Office on the web no longer opens in Internet Explorer or Microsoft Edge Legacy (EdgeHTML). Consequently, AppSource doesn't test Office on the web on these browsers. Office still supports these browsers for add-in runtimes, so if you think you've encountered a bug in how add-ins run in them, please create an issue for the [office-js](https://github.com/OfficeDev/office-js/issues) repository. For more information, see [Support older Microsoft browsers and Office versions](../develop/support-ie-11.md) and [Troubleshooting Microsoft Edge issues](../concepts/browsers-used-by-office-web-add-ins.md#troubleshoot-microsoft-edge-issues).
```

File: docs/testing/test-debug-office-add-ins.md
Comment (AlexJerabek) [inline docs/testing/test-debug-office-add-ins.md:31]: Some technicality phrasing.
```suggestion
Some Office versions on Windows still use the webview controls that come with Internet Explorer and Microsoft Edge Legacy. AppSource tests whether your add-in supports these browser controls. If your add-in doesn't support these browser controls, AppSource only issues a warning and doesn't reject your add-in. In this instance, we recommend configuring a graceful failure message on your add-in for a smoother user experience. For further guidance, see [Support older Microsoft browsers and Office versions](../develop/support-ie-11.md).
```

**Review by AlexJerabek — APPROVED** [review]
Approving with a couple more suggestions. Whether you take or leave them, I'm good with this being checked in without giving another review.

**Pull request merged**
